server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-        if multiplier <= 0:
-            raise ValueError("Alternative hypothesis is invalid!")
-
-        log_pvalue = (
-            self.sample_size * multiplier.ln()
-            - o1 * (1 - 1 / (2 * gamma)).ln()
-            - o2 * (1 - 1 / gamma).ln()
-            - u1 * (1 + 1 / (2 * gamma)).ln()
-            - u2 * (1 + 1 / gamma).ln()
-        )
-        pvalue = (log_pvalue).exp()
-        return float(np.min([float(pvalue), 1.0]))  # cast for the typechecker
-
-
-def maximize_fisher_combined_pvalue(
-    alpha: float,
-    contest: Contest,
-    bp_stratum: BallotPollingStratum,
-    cvr_stratum: BallotComparisonStratum,
-    winner: str,
-    loser: str,
-    stepsize: float = 0.05,
-) -> float:
-    """
-    Grid search to find the maximum P-value.
-    Find the smallest Fisher's combined statistic for P-values obtained
-    by testing two null hypotheses at level alpha using data X=(X1, X2).
-
-    This was largely taken from CORLA18.
-
-    Inputs:
-        alpha: the risk limit as a fraction
-        contest: the contest being audited
-        bp_stratum: a stratum object containing the ballot polling stratum
-        cvr_stratum: a stratum object containing the ballot comparison stratum
-        winner, loser: the winner-loser pair to maximuze the p-value over
-
-    Outputs:
-        a maximum p-value for the Fisher's combining function given the two
-        strata.
-    """
-    maximized_pvalue = 0.0
-    # find range of possible lambda
-    cvr_winner_votes = cvr_stratum.vote_totals[winner]
-    bp_winner_votes = bp_stratum.vote_totals[winner]
-
-    cvr_loser_votes = cvr_stratum.vote_totals[loser]
-    bp_loser_votes = bp_stratum.vote_totals[loser]
-
-    V = cvr_winner_votes - cvr_loser_votes + bp_winner_votes - bp_loser_votes
-
-    # The election is tied
-    if V == 0:
-        return 1.0
-
-    lambda_lower = (
-        np.amax(
-            [
-                cvr_winner_votes - cvr_loser_votes - cvr_stratum.num_ballots,
-                V - (bp_winner_votes - bp_loser_votes + bp_stratum.num_ballots),
-            ]
-        )
-        / V
-    )
-    lambda_upper = (
-        np.amax(
-            [
-                cvr_winner_votes - cvr_loser_votes + cvr_stratum.num_ballots,
-                V - (bp_winner_votes - bp_loser_votes - bp_stratum.num_ballots),
-            ]
-        )
-        / V
-    )
-
-    sample = bravo.compute_cumulative_sample(bp_stratum.sample)
-    bp_sample_winner_votes = sample[winner]
-    bp_sample_loser_votes = sample[loser]
-
-    reported_margin = contest.candidates[winner] - contest.candidates[loser]
-
-    Wn = bp_sample_winner_votes
-    Ln = bp_sample_loser_votes
-    Un = bp_stratum.sample_size - bp_sample_winner_votes - bp_sample_loser_votes
-    assert Wn >= 0, f"{Wn, Ln, Un}"
-    assert Ln >= 0, f"{Wn, Ln, Un}"
-    assert Un >= 0, f"{Wn, Ln, Un}"
-
-    T2 = (
-        lambda delta: 2
-        * bp_stratum.sample_size
```

Comment:
this should be `cvr_stratum.sample_size`

Reply:
Good catch. Idk why they called it T2, doesn't this correspond to T1 in the paper?

Reply:
I couldn't find T1 in the paper, except in the section that defined it as "Let Tj be the value of tp for the batch p selected in the jth draw," which seems different from this.

What I did find is that they refer to alternative test statistic functions (Liptak's or Tippet's) as T, so maybe T has something to do with that?

Reply:
T1 is in Appendix A on page 12 (as an expected value ET1). I think it's T for test statistic, 1 for stratum 1 (ballot comparison in this case). I dunno though.

Reply:
Yeah that's where I got the quote I cited from - just a few lines before that. Little t is for taint there.